### PR TITLE
Create and pass v0 flag to AOD

### PR DESCRIPTION
**Pull request by shahor02**

### CODE REVIEW

File: Detectors/AOD/src/AODProducerWorkflowSpec.cxx [hunk 1]
```diff
@@ -1176,7 +1177,7 @@ void AODProducerWorkflowDPL::fillSecondaryVertices(const o2::globaltracking::Rec
       collID = itemV->second;
     }
     if (posTableIdx != -1 and negTableIdx != -1 and collID != -1) {
-      v0Cursor(collID, posTableIdx, negTableIdx);
+      v0Cursor(collID, posTableIdx, negTableIdx, v0flags);
```
Comment (shahor02) [inline Detectors/AOD/src/AODProducerWorkflowSpec.cxx:1223]: @ddobrigk at the moment compilation fails here with 
```
/home/shahoian/alice/sw/SOURCES/O2/dev/0/Framework/Core/include/Framework/AnalysisHelpers.h: In instantiation of 'void o2::framework::WritingCursor<o2::soa::Table<C ...> >::operator()(T ...) [with T = {int, int, int, unsigned char}; PC = {o2::aod::v0::CollisionId, o2::aod::v0::PosTrackId, o2::aod::v0::NegTrackId}]':
/home/shahoian/alice/sw/SOURCES/O2/dev/0/Detectors/AOD/src/AODProducerWorkflowSpec.cxx:1180:15:   required from 'void o2::aodproducer::AODProducerWorkflowDPL::fillSecondaryVertices(const o2::globaltracking::RecoContainer&, V0CursorType&, CascadeCursorType&, Decay3BodyCursorType&) [with V0CursorType = o2::framework::Produces<o2::soa::Table<o2::soa::Index<0, -1>, o2::aod::v0::CollisionId, o2::aod::v0::PosTrackId, o2::aod::v0::NegTrackId> >; CascadeCursorType = o2::framework::Produces<o2::soa::Table<o2::soa::Index<0, -1>, o2::aod::cascade::CollisionId, o2::aod::cascade::V0Id, o2::aod::cascade::BachelorId> >; Decay3bodyCursorType = o2::framework::Produces<o2::soa::Table<o2::soa::Index<0, -1>, o2::aod::decay3body::CollisionId, o2::aod::decay3body::Track0Id, o2::aod::decay3body::Track1Id, o2::aod::decay3body::Track2Id> >]'
/home/shahoian/alice/sw/SOURCES/O2/dev/0/Detectors/AOD/src/AODProducerWorkflowSpec.cxx:2117:24:   required from here
/home/shahoian/alice/sw/SOURCES/O2/dev/0/Framework/Core/include/Framework/AnalysisHelpers.h:48:33: error: static assertion failed: Argument number mismatch
   48 |     static_assert(sizeof...(PC) == sizeof...(T), "Argument number mismatch");

```

**Comment by ddobrigk** [comment]
Thanks a lot, @shahor02 , indeed now the converter is merged so we can switch to `V0s_002` and finish this!

**Comment by shahor02** [comment]
@ddobrigk did a test on 2 TFs, svertexer reports
```
Found 24988 V0s (19313445 fits), 629 cascades (622716 fits), 28009 3-body decays (7136737 fits), 89 strange tracks. Timing: CPU: 9.66 Real: 5.03 s
Found 25401 V0s (18288144 fits), 1011 cascades (688061 fits), 31994 3-body decays (7711925 fits), 72 strange tracks. Timing: CPU: 6.43 Real: 1.74 s
```
and `O2v0_002->fV0Type` shows (0: v0 saved for the only due to the cascade, 1: standalone v0, 3: standard v0 with photon-only test done due to the presence of TPC-only track):

v0flag

To me looks reasonable. The only doubt I have is: as far as I understand, in the old AODs the fV0Type will be set to 0, so all v0s will look like saved for the cascade only. Can avoid this with some defaults or shall I swap the bits in the AODProducer to have "0: standard v0,  1: photon v0, 2: v0 for cascade" ?

**Comment by ddobrigk** [comment]
> @ddobrigk did a test on 2 TFs, svertexer reports
> 
> ```
> Found 24988 V0s (19313445 fits), 629 cascades (622716 fits), 28009 3-body decays (7136737 fits), 89 strange tracks. Timing: CPU: 9.66 Real: 5.03 s
> Found 25401 V0s (18288144 fits), 1011 cascades (688061 fits), 31994 3-body decays (7711925 fits), 72 strange tracks. Timing: CPU: 6.43 Real: 1.74 s
> ```
> 
> and `O2v0_002->fV0Type` shows (0: v0 saved for the only due to the cascade, 1: standalone v0, 3: standard v0 with photon-only test done due to the presence of TPC-only track):
> 
> v0flag
> 
> To me looks reasonable. The only doubt I have is: as far as I understand, in the old AODs the fV0Type will be set to 0, so all v0s will look like saved for the cascade only. Can avoid this with some defaults or shall I swap the bits in the AODProducer to have "0: standard v0, 1: photon v0, 2: v0 for cascade" ?

Hi @shahor02, no, this is unnecessary here because the default in case this isn't in the AO2D (i.e. for old AO2Ds) is defined in the V0 converter inside O2Physics and isn't necessarily zero. At the moment I set the default to 255, but actually now that we also incorporated the photons, I should set it simply to 1 to flag everything as a standard V0.

**Comment by shahor02** [comment]
@ddobrigk the fullCI fails in the analysis, could you please check? 
We should try to merge this PR this afternoon

**Comment by ddobrigk** [comment]
> @ddobrigk the fullCI fails in the analysis, could you please check? We should try to merge this PR this afternoon

Hi Ruben, yes, see here: https://github.com/AliceO2Group/O2Physics/pull/4156

**Comment by shahor02** [comment]
tested locally, CIs failing due the fix in the O2Physics not merged yet.
Merging.

**Pull request merged**